src/components/sequencing/SequenceEditor.svelte

```diff
@@ -274,6 +285,77 @@
     }
   }
 
+  const blockMark = Decoration.mark({ class: 'cm-block-match' });
+
+  const blockTheme = EditorView.baseTheme({
+    '.cm-block-match': {
+      outline: '1px dashed',
+    },
+  });
+
+  function highlightBlock(viewUpdate: ViewUpdate) {
+    const tree = syntaxTree(viewUpdate.state);
+    // Command Node includes trailing newline and white space, move to next command
+    const selectionLine = viewUpdate.state.doc.lineAt(viewUpdate.state.selection.asSingle().main.from);
+    const leadingWhiteSpaceLength = selectionLine.text.length - selectionLine.text.trimStart().length;
+    const updatedSelectionNode = tree.resolveInner(selectionLine.from + leadingWhiteSpaceLength, 1);
+    const stemNode = getNearestAncestorNodeOfType(updatedSelectionNode, [TOKEN_COMMAND])?.getChild('Stem');
+
+    if (!stemNode || !isBlockCommand(viewUpdate.state.sliceDoc(stemNode.from, stemNode.to))) {
+      return [];
+    }
+
+    const blocks = computeBlocks(viewUpdate.state);
+    if (!blocks) {
+      return [];
+    }
+
+    const pairs = Object.values(blocks);
+    const matchedNodes: SyntaxNode[] = [stemNode];
+
+    // when cursor on end -- select else and if
+    let current: SyntaxNode | undefined = stemNode;
+    while (current) {
+      current = pairs.find(block => block.end?.from === current!.from)?.start;
+      if (current) {
+        matchedNodes.push(current);
+      }
+    }
+
+    // when cursor on if -- select else and end
+    current = stemNode;
+    while (current) {
+      current = pairs.find(block => block.start?.from === current!.from)?.end;
+      if (current) {
+        matchedNodes.push(current);
+      }
+    }
+
+    return matchedNodes;
+  }
+
+  const blockHighlighter = ViewPlugin.fromClass(
+    class {
+      decorations: DecorationSet;
+      constructor(_view: EditorView) {
+        this.decorations = Decoration.none;
+      }
+      update(update: ViewUpdate) {
+        if (update.selectionSet || update.docChanged || update.viewportChanged) {
+          const blocks = highlightBlock(update);
+          this.decorations = Decoration.set(
+            // codemirror requires marks to be in sorted order
+            blocks.sort((a, b) => a.from - b.from).map(block => blockMark.range(block.from, block.to)),
+          );
+          return this.decorations;
+        }
+      }
+    },
+    {
+      decorations: v => v.decorations,
```

Comment:
Do you mind renaming `v` to something more explicit so it's easier to understand what type of object it is?

Reply:
Sorry, would `view` be more clear as to what it is? `spec` seems a little ambiguous within the context of aerie.